### PR TITLE
Change propagate boolean values 

### PR DESCRIPTION
**Pull request by chandrashritii**
**Description**

Changed propagate boolean values to reflect child job results in top level pipeline

This PR fixes https://github.com/adoptium/ci-jenkins-pipelines/issues/234

Signed-off-by: Shriti Chandra [shchandr@redhat.com](mailto:shchandr@redhat.com)

### CODE REVIEW

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -828,7 +828,7 @@ class Builder implements Serializable {
                             // Pass down DEFAULTS_JSON
                             buildJobParams.add(['$class': 'TextParameterValue', name: 'DEFAULTS_JSON', value: JsonOutput.prettyPrint(JsonOutput.toJson(DEFAULTS_JSON)) ])
 
-                            def downstreamJob = context.build job: downstreamJobName, propagate: false, parameters: buildJobParams
+                            def downstreamJob = context.build job: downstreamJobName, propagate: true, parameters: buildJobParams
```
Comment (andrew-m-leonard) [inline pipelines/build/common/build_base_file.groovy:831]: This one needs to stay "false", as the PARAM.propagateFailures check in the if (downstreamJob.getResult() == 'SUCCESS') ... else...
handles it

**Comment by andrew-m-leonard** [comment]
@chandrashritii Please can you update the comment in the pipeline_job_template and release_pipeline_job_template:
```
        booleanParam('propagateFailures', propagateFailures, 'If true, a failure of <b>ANY</b> downstream build (but <b>NOT</b> test) will cause the whole build to fail')
```
remove the (but NOT test)

**Comment by andrew-m-leonard** [comment]
Test build: https://ci.adoptium.net/job/build-scripts/job/openjdk17-pipeline/634/

**Comment by andrew-m-leonard** [comment]
@chandrashritii The test job didn't quite do what I expected: https://ci.adoptium.net/job/build-scripts/job/openjdk17-pipeline/634/
So the pipeline -> build job -> sanity.tests
In the above test, one of the sanity.tests was Unstable(Yellow), this then now means the build-job result is now Unstable, this then means the Base pipeline code here: https://github.com/adoptium/ci-jenkins-pipelines/blob/9793cd51470be282fb8152088a90f088c4bd57f6/pipelines/build/common/build_base_file.groovy#L833
Will not now copy the artifacts.. 

So this relates to the comments in the issue: https://github.com/adoptium/ci-jenkins-pipelines/issues/234#issue-1114230663

I think we should remove the "if (downstreamJob.getResult() == 'SUCCESS') {" here: https://github.com/adoptium/ci-jenkins-pipelines/blob/9793cd51470be282fb8152088a90f088c4bd57f6/pipelines/build/common/build_base_file.groovy#L833
and ALWAYS try and copy artifacts, probably put the whole section that is in the if, into a try/catch.

Then following that do the result check to propagate the result if required something like:
```
// If build job not successful and we need to propagate failures, then set the pipeline job result
if (downstreamJob.getResult() != 'SUCCESS' && propagateFailures) {
   if (downstreamJob.getResult() == 'UNSTABLE' && currentBuild.result == 'SUCCESS') {
       // Build job Unstable and current result is Success, set to Unstable
       currentBuild.result = 'UNSTABLE'
   else if (currentBuild.result != 'FAILURE') {
       // If current result is not Failure, then propagate this build job result
       currentBuild.result = downstreamJob.getResult()
   }
   context.error("Downstream build ${downstreamJobName} reported resuolt : "+downstreamJob.getResult())
}
```

**Comment by chandrashritii** [comment]
Thanks for the review! @andrew-m-leonard Got the unstable build-job result covered in a try-catch block. Are any other changes needed?

File: pipelines/build/common/build_base_file.groovy [hunk 2]
```diff
@@ -830,60 +830,66 @@ class Builder implements Serializable {
...
+                                                }
+                                            } catch (FlowInterruptedException e) {
+                                                throw new Exception("[ERROR] Archive artifact timeout (${pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT} HOURS) for ${downstreamJobName}has been reached. Exiting...")
                                             }
-                                        } catch (FlowInterruptedException e) {
-                                            throw new Exception("[ERROR] Archive artifact timeout (${pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT} HOURS) for ${downstreamJobName}has been reached. Exiting...")
                                         }
                                     }
+                                    context.println '[NODE SHIFT] OUT OF CONTROLLER NODE!'
+                                } catch (propagateFailures) {
```
Comment (andrew-m-leonard) [inline pipelines/build/common/build_base_file.groovy:882]: This catch needs to be:
```
} catch (FlowInterruptedException e) {
 ...log exception..
set currentBuild.result to "UNSTABLE" if not already set to "FAILURE", to indicate copy artifact problem
}
```

File: pipelines/build/common/build_base_file.groovy [hunk 3]
```diff
@@ -830,60 +830,66 @@ class Builder implements Serializable {
...
+                                    if (downstreamJob.getResult() != 'SUCCESS') {
```
Comment (andrew-m-leonard) [inline pipelines/build/common/build_base_file.groovy:883]: This if wants to be outside the copyArtifact catch

**Comment by andrew-m-leonard** [comment]
These changes have been incorporated and merged under : https://github.com/adoptium/ci-jenkins-pipelines/pull/649

**Pull request closed without merging**